## Goal: Merge all six months parquet files into one before starting with the cleaning process.

### Importing necessary libraries and merging all 6 into 1 using loop.

In [14]:
import pandas as pd
import glob

# Get first 6 parquet files
files = sorted(glob.glob("yellow_tripdata_2025-*.parquet"))[:6]

# Read and combine
df = pd.concat(
    [pd.read_parquet(file) for file in files],
    ignore_index=True
)

#### After merging checking info of the merged dataset

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24083384 entries, 0 to 24083383
Data columns (total 20 columns):
 #   Column                 Dtype         
---  ------                 -----         
 0   VendorID               int32         
 1   tpep_pickup_datetime   datetime64[us]
 2   tpep_dropoff_datetime  datetime64[us]
 3   passenger_count        float64       
 4   trip_distance          float64       
 5   RatecodeID             float64       
 6   store_and_fwd_flag     object        
 7   PULocationID           int32         
 8   DOLocationID           int32         
 9   payment_type           int64         
 10  fare_amount            float64       
 11  extra                  float64       
 12  mta_tax                float64       
 13  tip_amount             float64       
 14  tolls_amount           float64       
 15  improvement_surcharge  float64       
 16  total_amount           float64       
 17  congestion_surcharge   float64       
 18  Airport_fee         

This dataset has 24083384 rows and 20 columns with following datatypes: datetime64[us](2), float64(13), int32(3), int64(1), object(1)
and 3.3+ GB memory usage.

#### Viewing random 20 rows of the dataset

In [53]:
df.sample(20)

,VendorID,tpep_pickup_datetime,tpep_dropoff_datetime,passenger_count,trip_distance,RatecodeID,store_and_fwd_flag,PULocationID,DOLocationID,payment_type,fare_amount,extra,mta_tax,tip_amount,tolls_amount,improvement_surcharge,total_amount,congestion_surcharge,Airport_fee,cbd_congestion_fee
15409660,2,2025-05-02 23:43:10,2025-05-02 23:52:00,1.0,1.56,1.0,N,164,79,2,10.700000,1.00,0.5,0.00,0.00,1.0,16.450001,2.5,0.00,0.75
3305287,2,2025-01-23 22:33:25,2025-01-23 22:38:03,NaN,0.85,NaN,None,90,186,0,-4.750000,0.00,0.5,0.00,0.00,1.0,3.390000,NaN,NaN,0.75
18384421,2,2025-05-30 13:10:04,2025-05-30 13:21:01,2.0,1.01,1.0,N,237,236,1,11.400000,0.00,0.5,3.08,0.00,1.0,18.480000,2.5,0.00,0.00
14028373,2,2025-04-27 09:07:17,2025-04-27 09:10:21,1.0,1.11,1.0,N,48,68,1,6.500000,0.00,0.5,2.25,0.00,1.0,13.500000,2.5,0.00,0.75
3929752,1,2025-02-05 17:32:44,2025-02-05 17:43:16,1.0,1.70,1.0,N,143,151,3,10.700000,5.00,0.5,0.00,0.00,1.0,17.200001,2.5,0.00,0.00
22837160,2,2025-06-30 17:13:17,2025-06-30 17:17:11,1.0,0.48,1.0,N,141,236,1,5.800000,2.50,0.5,2.46,0.00,1.0,14.760000,2.5,0.00,0.00
10672812,1,2025-03-14 08:54:36,2025-03-14 08:59:40,NaN,0.60,NaN,None,262,236,0,9.080000,0.00,0.5,0.00,0.00,1.0,13.080000,NaN,NaN,0.00
11152108,2,2025-03-30 13:46:16,2025-03-30 13:55:48,NaN,1.95,NaN,None,100,68,0,14.060000,0.00,0.5,3.76,0.00,1.0,22.570000,NaN,NaN,0.75
5215872,2,2025-02-18 20:27:26,2025-02-18 20:46:10,1.0,5.33,1.0,N,107,13,1,26.100000,1.00,0.5,1.59,0.00,1.0,33.439999,2.5,0.00,0.75
8461986,2,2025-03-14 13:37:02,2025-03-14 13:49:22,1.0,1.34,1.0,N,68,161,1,12.100000,0.00,0.5,3.37,0.00,1.0,20.219999,2.5,0.00,0.75


#### Converting and downloading the 6 files merged into 1

In [54]:
df.to_parquet('yellow_taxi_6_months.parquet')